Some of the tests/test_main.py::test_main_requires* tests seem unstable

### Current Behavior

I noticed that sometimes `nox -e tests-3.14` failed locally on my branch (#1129). I don't know how to properly reproduce it, and after some more tries I also was able to reproduce this on `main` and when only running `nox -e tests-3.14 -- tests/test_main.py`.

I've seen these tests fail:
* `tests/test_main.py::test_main_requries_modern_param[w]`
* `tests/test_main.py::test_main_requires[sessions3-expected_order3]`
* `tests/test_main.py::test_main_requires[sessions4-expected_order4]`


### Expected Behavior

Tests pass

### Steps To Reproduce

I was able to reproduce this by running `nox -e tests-3.14 -- tests/test_main.py` multiple times, with shorter and longer delays inbetween. Usually it didn't happen after only a short delay. On my machine, pytest is using 16 workers, so my guess is that it is a concurrency problem.

### Environment

```Markdown
- OS: Arch Linux
- Python: Python 3.14.6, pytest-9.1.1, pluggy-1.6.0
- Nox: `main` branch
```

### Anything else?

Here is a run with two failures on `main`:
```
$ nox -e tests-3.14 -- tests/test_main.py
nox > Running session tests-3.14
nox > Creating virtual environment (uv) using python3.14 in .nox/tests-3-14
nox > /usr/bin/uv pip install 'coverage[toml]>=7.10.3' 'pytest>=7.4; python_version>='"'"'3.12'"'"'' 'pytest>=7; python_version<'"'"'3.12'"'"'' 'pytest-xdist>=3' uv
nox > uv pip install '-e.[tox-to-nox,pbs]'
nox > coverage erase
nox > coverage run -m pytest --numprocesses=auto -m 'not conda' tests/test_main.py
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.14.6, pytest-9.1.1, pluggy-1.6.0
rootdir: /path/to/nox
configfile: pyproject.toml
plugins: anyio-4.14.1, xdist-3.8.0
16 workers [133 items]    
........................s.......................................................................................s..........F.F.......                                [100%]
================================================================================= FAILURES =================================================================================
____________________________________________________________________ test_main_requries_modern_param[w] ____________________________________________________________________
[gw14] linux -- Python 3.14.6 /path/to/nox/.nox/tests-3-14/bin/python3

run_nox = <function run_nox.<locals>._run_nox at 0x7fefe76c6a30>, session = 'w'

    @pytest.mark.parametrize("session", ["w", "u"])
    def test_main_requries_modern_param(
        run_nox: Callable[..., tuple[Any, Any, Any]],
        session: str,
    ) -> None:
        noxfile = os.path.join(RESOURCES, "noxfile_requires.py")
        returncode, _, _stderr = run_nox(f"--noxfile={noxfile}", f"--session={session}")
>       assert returncode == 0
E       assert 1 == 0

/path/to/nox/tests/test_main.py:644: AssertionError
---------------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------------
SESSION_INFO nox:logger.py:90 Running session u(django='1.9')
INFO     nox:virtualenv.py:948 Creating virtual environment (virtualenv) using python3 in .nox/u-django-1-9
SUCCESS  nox:logger.py:94 Session u(django='1.9') was successful.
SESSION_INFO nox:logger.py:90 Running session u(django='2.0')
INFO     nox:virtualenv.py:948 Creating virtual environment (virtualenv) using python3 in .nox/u-django-2-0
ERROR    nox:command.py:201 Command /path/to/nox/.nox/tests-3-14/bin/python3 -m virtualenv /path/to/nox/tests/resources/.nox/u-django-2-0 --no-periodic-update failed with exit code 1:
ERROR    nox:command.py:206 RuntimeError: failed to build image pip because:
Traceback (most recent call last):
  File "/path/to/nox/.nox/tests-3-14/lib/python3.14/site-packages/virtualenv/seed/embed/via_app_data/via_app_data.py", line 71, in _install
    installer.install(creator.interpreter.version_info)  # ty: ignore[invalid-argument-type]
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/to/nox/.nox/tests-3-14/lib/python3.14/site-packages/virtualenv/seed/embed/via_app_data/pip_install/base.py", line 59, in install
    self._uninstall_previous_version()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/path/to/nox/.nox/tests-3-14/lib/python3.14/site-packages/virtualenv/seed/embed/via_app_data/pip_install/base.py", line 185, in _uninstall_previous_version
    self._uninstall_dist(existing_dist)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/path/to/nox/.nox/tests-3-14/lib/python3.14/site-packages/virtualenv/seed/embed/via_app_data/pip_install/base.py", line 213, in _uninstall_dist
    path.unlink()
    ~~~~~~~~~~~^^
  File "/usr/lib/python3.14/pathlib/__init__.py", line 1042, in unlink
    os.unlink(self)
    ~~~~~~~~~^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/path/to/nox/tests/resources/.nox/u-django-2-0/lib/python3.14/site-packages/pip/_internal/locations/_sysconfig.py'


ERROR    nox:sessions.py:1323 Session u(django='2.0') failed.
SESSION_INFO nox:logger.py:90 Running session w
ERROR    nox:sessions.py:1323 Session w aborted: Prerequisite session u(django='2.0') was not successful.
SESSION_INFO nox:logger.py:90 Ran 3 sessions in a second:
SUCCESS  nox:logger.py:94 * u(django='1.9'): success
ERROR    nox:sessions.py:1323 * u(django='2.0'): failed
ERROR    nox:sessions.py:1323 * w: aborted
______________________________________________________________ test_main_requires[sessions3-expected_order3] _______________________________________________________________
[gw7] linux -- Python 3.14.6 /path/to/nox/.nox/tests-3-14/bin/python3

run_nox = <function run_nox.<locals>._run_nox at 0x7ff4d9386610>, sessions = ('v',), expected_order = ("u(django='1.9')", "u(django='2.0')", 'v')

    @pytest.mark.parametrize(
        ("sessions", "expected_order"),
        [
            (("g", "a", "d"), ("b", "c", "h", "g", "a", "e", "d")),
            pytest.param(("m",), ("k-3.10", "k-3.11", "m"), marks=py310py311),
            pytest.param(("n",), ("k-3.11", "n"), marks=py310py311),
            (("v",), ("u(django='1.9')", "u(django='2.0')", "v")),
            (("w",), ("u(django='1.9')", "u(django='2.0')", "w")),
        ],
    )
    def test_main_requires(
        run_nox: Callable[..., tuple[Any, Any, Any]],
        sessions: tuple[str, ...],
        expected_order: tuple[str, ...],
    ) -> None:
        noxfile = os.path.join(RESOURCES, "noxfile_requires.py")
        returncode, stdout, _ = run_nox(f"--noxfile={noxfile}", "--sessions", *sessions)
>       assert returncode == 0
E       assert 1 == 0

/path/to/nox/tests/test_main.py:596: AssertionError
---------------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------------
SESSION_INFO nox:logger.py:90 Running session u(django='1.9')
INFO     nox:virtualenv.py:948 Creating virtual environment (virtualenv) using python3 in .nox/u-django-1-9
ERROR    nox:command.py:201 Command /path/to/nox/.nox/tests-3-14/bin/python3 -m virtualenv /path/to/nox/tests/resources/.nox/u-django-1-9 --no-periodic-update failed with exit code 1:
ERROR    nox:command.py:206 RuntimeError: failed to build image pip because:
Traceback (most recent call last):
  File "/path/to/nox/.nox/tests-3-14/lib/python3.14/site-packages/virtualenv/seed/embed/via_app_data/via_app_data.py", line 71, in _install
    installer.install(creator.interpreter.version_info)  # ty: ignore[invalid-argument-type]
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/to/nox/.nox/tests-3-14/lib/python3.14/site-packages/virtualenv/seed/embed/via_app_data/pip_install/base.py", line 59, in install
    self._uninstall_previous_version()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/path/to/nox/.nox/tests-3-14/lib/python3.14/site-packages/virtualenv/seed/embed/via_app_data/pip_install/base.py", line 185, in _uninstall_previous_version
    self._uninstall_dist(existing_dist)
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^
  File "/path/to/nox/.nox/tests-3-14/lib/python3.14/site-packages/virtualenv/seed/embed/via_app_data/pip_install/base.py", line 213, in _uninstall_dist
    path.unlink()
    ~~~~~~~~~~~^^
  File "/usr/lib/python3.14/pathlib/__init__.py", line 1042, in unlink
    os.unlink(self)
    ~~~~~~~~~^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/path/to/nox/tests/resources/.nox/u-django-1-9/lib/python3.14/site-packages/pip/_internal/resolution/resolvelib/requirements.py'


ERROR    nox:sessions.py:1323 Session u(django='1.9') failed.
SESSION_INFO nox:logger.py:90 Running session u(django='2.0')
INFO     nox:virtualenv.py:948 Creating virtual environment (virtualenv) using python3 in .nox/u-django-2-0
SUCCESS  nox:logger.py:94 Session u(django='2.0') was successful.
SESSION_INFO nox:logger.py:90 Running session v
ERROR    nox:sessions.py:1323 Session v aborted: Prerequisite session u(django='1.9') was not successful.
SESSION_INFO nox:logger.py:90 Ran 3 sessions in a second:
ERROR    nox:sessions.py:1323 * u(django='1.9'): failed
SUCCESS  nox:logger.py:94 * u(django='2.0'): success
ERROR    nox:sessions.py:1323 * v: aborted
========================================================================= short test summary info ==========================================================================
SKIPPED [2] tests/test_main.py:579: Python 3.10 and 3.11 required
FAILED tests/test_main.py::test_main_requries_modern_param[w] - assert 1 == 0
FAILED tests/test_main.py::test_main_requires[sessions3-expected_order3] - assert 1 == 0
================================================================= 2 failed, 129 passed, 2 skipped in 4.39s =================================================================
nox > Command coverage run -m pytest --numprocesses=auto -m 'not conda' tests/test_main.py failed with exit code 1
nox > Session tests-3.14 failed.
```

Here another run with one failure on my branch:
```
=========================================================================== test session starts ============================================================================
platform linux -- Python 3.14.6, pytest-9.1.1, pluggy-1.6.0
rootdir: /path/to/nox
configfile: pyproject.toml
plugins: anyio-4.14.1, xdist-3.8.0
16 workers [133 items]    
.............................s.................................................................s.......................F.............                                [100%]
================================================================================= FAILURES =================================================================================
______________________________________________________________ test_main_requires[sessions4-expected_order4] _______________________________________________________________
[gw9] linux -- Python 3.14.6 /path/to/nox/.nox/tests-3-14/bin/python3

run_nox = <function run_nox.<locals>._run_nox at 0x7ff05114c5c0>, sessions = ('w',), expected_order = ("u(django='1.9')", "u(django='2.0')", 'w')

    @pytest.mark.parametrize(
        ("sessions", "expected_order"),
        [
            (("g", "a", "d"), ("b", "c", "h", "g", "a", "e", "d")),
            pytest.param(("m",), ("k-3.10", "k-3.11", "m"), marks=py310py311),
            pytest.param(("n",), ("k-3.11", "n"), marks=py310py311),
            (("v",), ("u(django='1.9')", "u(django='2.0')", "v")),
            (("w",), ("u(django='1.9')", "u(django='2.0')", "w")),
        ],
    )
    def test_main_requires(
        run_nox: Callable[..., tuple[Any, Any, Any]],
        sessions: tuple[str, ...],
        expected_order: tuple[str, ...],
    ) -> None:
        noxfile = os.path.join(RESOURCES, "noxfile_requires.py")
        returncode, stdout, _ = run_nox(f"--noxfile={noxfile}", "--sessions", *sessions)
>       assert returncode == 0
E       assert 1 == 0

/path/to/nox/tests/test_main.py:596: AssertionError
---------------------------------------------------------------------------- Captured log call -----------------------------------------------------------------------------
SESSION_INFO nox:logger.py:90 Running session u(django='1.9')
INFO     nox:virtualenv.py:948 Creating virtual environment (virtualenv) using python3 in .nox/u-django-1-9
SUCCESS  nox:logger.py:94 Session u(django='1.9') was successful.
SESSION_INFO nox:logger.py:90 Running session u(django='2.0')
INFO     nox:virtualenv.py:948 Creating virtual environment (virtualenv) using python3 in .nox/u-django-2-0
ERROR    nox:command.py:204 Command /path/to/nox/.nox/tests-3-14/bin/python3 -m virtualenv /path/to/nox/tests/resources/.nox/u-django-2-0 --no-periodic-update failed with exit code 1:
ERROR    nox:command.py:209 RuntimeError: failed to build image pip because:
Traceback (most recent call last):
  File "/path/to/nox/.nox/tests-3-14/lib/python3.14/site-packages/virtualenv/seed/embed/via_app_data/via_app_data.py", line 71, in _install
    installer.install(creator.interpreter.version_info)  # ty: ignore[invalid-argument-type]
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/path/to/nox/.nox/tests-3-14/lib/python3.14/site-packages/virtualenv/seed/embed/via_app_data/pip_install/base.py", line 59, in install
    self._uninstall_previous_version()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/path/to/nox/.nox/tests-3-14/lib/python3.14/site-packages/virtualenv/seed/embed/via_app_data/pip_install/base.py", line 181, in _uninstall_previous_version
    in_folders = chain.from_iterable([i.iterdir() for i in (self._creator.purelib, self._creator.platlib)])
                                      ~~~~~~~~~^^
  File "/usr/lib/python3.14/pathlib/__init__.py", line 836, in iterdir
    with os.scandir(root_dir) as scandir_it:
         ~~~~~~~~~~^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/path/to/nox/tests/resources/.nox/u-django-2-0/lib/python3.14/site-packages'


ERROR    nox:sessions.py:1323 Session u(django='2.0') failed.
SESSION_INFO nox:logger.py:90 Running session w
ERROR    nox:sessions.py:1323 Session w aborted: Prerequisite session u(django='2.0') was not successful.
SESSION_INFO nox:logger.py:90 Ran 3 sessions:
SUCCESS  nox:logger.py:94 * u(django='1.9'): success
ERROR    nox:sessions.py:1323 * u(django='2.0'): failed
ERROR    nox:sessions.py:1323 * w: aborted
========================================================================= short test summary info ==========================================================================
SKIPPED [2] tests/test_main.py:579: Python 3.10 and 3.11 required
FAILED tests/test_main.py::test_main_requires[sessions4-expected_order4] - assert 1 == 0
================================================================= 1 failed, 130 passed, 2 skipped in 4.35s =================================================================
nox > Command coverage run -m pytest --numprocesses=auto -m 'not conda' tests/test_main.py failed with exit code 1
nox > Session tests-3.14 failed.
```

My guess is that two threads / processes in parallel run two tests that use the same directory in `.nox`, and this causes problems when two `pip` instances try to modify the same `site-packages/pip/` directory concurrently.

--- a/tests/test_main.py
+++ b/tests/test_main.py
@@ -493,9 +493,16 @@ def test_main_session_with_names(
 
 @pytest.fixture
 def run_nox(
-    capsys: pytest.CaptureFixture[str], monkeypatch: pytest.MonkeyPatch
+    capsys: pytest.CaptureFixture[str],
+    monkeypatch: pytest.MonkeyPatch,
+    tmp_path: Path,
 ) -> Callable[..., tuple[int, str, str]]:
     def _run_nox(*args: str) -> tuple[int, str, str]:
+        # Give each invocation its own envdir so sessions that build real
+        # virtualenvs don't collide with other xdist workers sharing the
+        # noxfile's default ``.nox`` directory.
+        if not any(arg.startswith("--envdir") for arg in args):
+            args = (*args, "--envdir", str(tmp_path / ".nox"))
         monkeypatch.setattr(sys, "argv", ["nox", *args])
 
         with mock.patch("sys.exit") as sys_exit:

